partner_operating_unit/tests/test_res_partner_category_operating_unit.py

```diff
@@ -0,0 +1,109 @@
+        super(TestResPartnerCategoryOperatingUnit, self).setUp()
+
+        self.ResUsers = self.env['res.users']
+        self.ResPartnerCategory = self.env['res.partner.category']
+        self.OperatingUnit = self.env['operating.unit']
+
+        # Companies.
+        self.company = self.env.ref('base.main_company')
+
+        # Groups.
+        self.group_partner_manager = self.env.ref('base.group_partner_manager')
+
+        # Operating Units.
+        self.opunit1 = self.OperatingUnit.create({
+            'name': 'Operating Unit #1',
+            'code': 'OP#1',
+            'partner_id': self.env.ref('base.main_partner').id,
+        })
+        self.opunit2 = self.OperatingUnit.create({
+            'name': 'Operating Unit #2',
+            'code': 'OP#2',
+            'partner_id': self.env.ref('base.main_partner').id,
+        })
+
+        # Partner categories
+        self.partner_categ1 = self._create_partner_category('Categ. #1')
+        self.partner_categ2 = self._create_partner_category('Categ. #2')
+        self.partner_categ3 = self._create_partner_category('Categ. #3')
+
+        # Sets the operating-units for each partner category.
+        self.partner_categ1.operating_unit_ids = [(6, False, [self.opunit1.id])]
+        self.partner_categ2.operating_unit_ids = [(6, False, [self.opunit2.id])]
+        self.partner_categ3.operating_unit_ids = [(6, False, [self.opunit1.id,
+                                                              self.opunit2.id])]
+
+        # Users.
+        self.user1_id = self._create_user('user_1', self.group_partner_manager,
+                                          [self.opunit1])
+        self.user2_id = self._create_user('user_2', self.group_partner_manager,
+                                          [self.opunit2])
+
+    def _create_partner_category(self, name):
+        """ Create a partner category.
+        """
+        return self.ResPartnerCategory.create({
+            'name': name,
+            'active': True,
+        })
+
+    def _create_user(self, login, groups, operating_units):
+        """ Create a user.
+        """
+        user = self.ResUsers.with_context({'no_reset_password': True}).create({
+            'name': 'Demo User',
+            'login': login,
+            'password': 'demo',
+            'email': '{0}@yourcompany.com'.format(login),
+            'operating_unit_ids': [(4, ou.id) for ou in operating_units],
+            'groups_id': [(6, 0, groups.ids)],
+        })
+        return user.id
+
+    def test_operating_units_for_partner_categories(self):
+        """ Test Security of Partner's Category Operating Unit
+        """
+        # User 1 is only assigned to Operating Unit 1, and can see all
+        # contact's tags having Operating Unit 1.
+        partner_categs = self.ResPartnerCategory.sudo(self.user1_id).search([
+            ('operating_unit_ids', 'in', self.opunit1.id),
+        ])
+        self.assertEqual(set(partner_categs), {self.partner_categ1,
```

Comment:
What is this creating? They are singleton recordsets. Is this creating a dictionary with them?
{self.partner_categ1, self.partner_categ3}

Why not doing set([self.partner_categ1.id, self.partner_categ3.id])?

Reply:
Same comment for two cases below

Reply:
It's creating a set. Using ```{}``` to create a set is 3 times more efficient than creating a list and then creating a set from it as in ```set([])```. Look:

```
>>> from timeit import timeit
>>> timeit("set([1,2,3])")
0.2715098049993685
>>> timeit("set([1,2,3])")
0.27455895600178337
>>> timeit("set([1,2,3])")
0.27554412400058936
>>> timeit("{1,2,3}")
0.1102297079996788
>>> timeit("{1,2,3}")
0.1061255530003109
>>> timeit("{1,2,3}")
0.1083366439997917
```

But well, using this on a test is probably overkill. But anyway, the same way that lists are created with ```[]``` then sets are created with ```{}```.

Reply:
OK, good to know {} is also for sets, not just for dictionaries. Thanks